### PR TITLE
[CWS] display evaluated events in timeout errors in tests

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

The goal of this PR is to collect events going through the current ruleset, and displaying them when a test fails because of a timeout. This will help investigation of flaky tests by showing for example events that should have matched but whose fields are broken, not what is expected.

This feature is enabled only if the `functionaltests` build tag is enabled so that it doesn't impact the actual system probe build.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by YoannGh — APPROVED** [review]

File: pkg/security/secl/rules/ruleset.go [hunk 1]
```diff
@@ -674,6 +678,8 @@ func (rs *RuleSet) Evaluate(event eval.Event) bool {
 		}
 	}
 
+	rs.eventCollector.CollectEvent(rs, event, result)
```
Comment (modernplumbing) [inline pkg/security/secl/rules/ruleset.go:681]: I would like a comment here saying that this func is a no-op in non-functional test builds because it can be hard to follow build tags (unlike interfaces)

Comment (paulcacheux) [reply pkg/security/secl/rules/ruleset.go:683]: added in last commit

File: pkg/security/secl/rules/collected_events_functests.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+		Fields:     make(map[string]interface{}, len(rs.fields)),
+	}
+
+	for _, field := range rs.fields {
+		fieldEventType, err := event.GetFieldEventType(field)
+		if err != nil {
+			rs.logger.Errorf("failed to get event type for field %s: %v", field, err)
+		}
+
+		if fieldEventType != "*" && fieldEventType != eventType {
```
Comment (modernplumbing) [inline pkg/security/secl/rules/collected_events_functests.go:38]: fieldEventType == "*" or fieldEventType == eventType happen?

File: pkg/security/secl/rules/collected_events_regular.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+package rules
+
+import "github.com/DataDog/datadog-agent/pkg/security/secl/compiler/eval"
+
+type EventCollector struct {
+}
+
+func (ec *EventCollector) CollectEvent(rs *RuleSet, event eval.Event, result bool) {
+	// no-op
```
Comment (modernplumbing) [inline pkg/security/secl/rules/collected_events_regular.go:16]: is not having a return going to make linters mad?

**Review by modernplumbing — APPROVED** [review]
